src/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Activity.cs

```diff
@@ -343,7 +409,173 @@ public void Stop()
+                    }
+                }
+                return ref _traceId;
+            }
+        }
+
+        /// <summary>
+        /// If the parent Activity ID has the W3C format, this returns the ID for the SpanId part of the ParentId.  
+        /// Otherwise it returns a zero SpanId. 
+        /// </summary>
+        public ref readonly ActivitySpanId ParentSpanId
+        {
+            get
+            {
+                if (!_parentSpanIdSet && _parentId != null && IsW3CId(_parentId))
+                {
+                    _parentSpanId = new ActivitySpanId(((ReadOnlySpan<char>)_parentId).Slice(36, 16));
+                    _parentSpanIdSet = true;
+                }
+                return ref _parentSpanId;
+            }
+        }
+
+        /// <summary>
+        /// Returns the format for the ID.   
+        /// </summary>
+        public ActivityIdFormat IdFormat { get; private set; }
+
+        /* static state (configuration) */
+        /// <summary>
+        /// Activity tries to use the same format for IDs as its parent.
+        /// However if the activity has no parent, it has to do something.   
+        /// This determines the default format we use.  
+        /// </summary>
+        public static ActivityIdFormat DefaultIdFormat
+        {
+            get
+            {
+                if (s_DefaultIdFormat == ActivityIdFormat.Unknown)
+                    s_DefaultIdFormat = ActivityIdFormat.Hierarchical;
+                return s_DefaultIdFormat;
+            }
+            set
+            {
+                if (!(ActivityIdFormat.Hierarchical <= value && value <= ActivityIdFormat.W3C))
+                    throw new ArgumentException($"value must be a valid ActivityIDFormat value");
+                s_DefaultIdFormat = value;
+
+            }
+        }
+
+        /// <summary>
+        /// Normally if the ParentID is defined, the format of that is used to determine the
+        /// format used by the Activity.   However if ForceDefaultFormat is set to true, the
+        /// ID format will always be the DefaultIdFormat even if the ParentID is define and is
+        /// a different format. 
+        /// </summary>
+        public static bool ForceDefaultIdFormat { get; set; }
+
         #region private 
+        /// <summary>
+        /// Returns true if 'id' has the format of a WC3 id see https://w3c.github.io/trace-context
+        /// </summary>
+        private static bool IsW3CId(string id)
+        {
+            // A W3CId is  
+            //  * 2 chars Version
+            //  * 1 char - char
+            //  * 32 chars traceId
+            //  * 1 char - char 
+            //  * 16 chars spanId
+            //  * 1 char - char
+            //  * 2 chars flags 
+            //  = 55 chars (see https://w3c.github.io/trace-context)
+            // We require that all non-WC3IDs NOT start with a digit.  
+            return id.Length == 55 && char.IsDigit(id[0]);
```

Comment:
char.IsDigit checks for more than just '0'-'9'... is that ok?

Reply:
What does IsDigit do besides 0..9?   
Note that the goal here was a very fast way of looking at a string and separating valid Hierarchical IDs (which always being with | from W3C Ids (which always begin with 0-9).   It is better if it is as  'narrow' as possible, so if IsDigit returns true for more than 0..9 it is probably better to replace it with an explicit range check.

Reply:
> What does IsDigit do besides 0..9?

It factors in Unicode digits:
https://github.com/dotnet/coreclr/blob/99b0e1a1729d354d4444c8fbd4149df30f7f2940/src/System.Private.CoreLib/shared/System/Char.cs#L209-L216